File: Creacion API/APIRest2_23_24_Capas/APIRest2_23_24/src/main/java/es/luisherrero/apirest1/model/Equipo.java
package es.luisherrero.apirest1.model;

import java.io.Serializable;
import java.util.Date;
import jakarta.persistence.*;

@Entity
@Table(name="equipos")
public class Equipo implements Serializable{
	private static final long serialVersionUID=1;
	@Id
	private int id;

	private byte baja;

	private String descripcion;

	private String etiqueta;

	@Column(name="fecha_adquisicion")
	private Date fechaAdquisicion;

	private String marca;

	private String modelo;

	private int puesto;
	
	//@Enumerated(EnumType.STRING)
	@Column(name="tipo_equipo")
	private String tipoEquipo;

	@Column(name = "aula_num")
	private int aula;

	public Equipo() {
	}

	public int getId() {
		return this.id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public byte getBaja() {
		return this.baja;
	}

	public void setBaja(byte baja) {
		this.baja = baja;
	}

	public String getDescripcion() {
		return this.descripcion;
	}

	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	public String getEtiqueta() {
		return this.etiqueta;
	}

	public void setEtiqueta(String etiqueta) {
		this.etiqueta = etiqueta;
	}

	public Date getFechaAdquisicion() {
		return this.fechaAdquisicion;
	}

	public void setFechaAdquisicion(Date fechaAdquisicion) {
		this.fechaAdquisicion = fechaAdquisicion;
	}

	public String getMarca() {
		return this.marca;
	}

	public void setMarca(String marca) {
		this.marca = marca;
	}

	public String getModelo() {
		return this.modelo;
	}

	public void setModelo(String modelo) {
		this.modelo = modelo;
	}

	public int getPuesto() {
		return this.puesto;
	}

	public void setPuesto(int puesto) {
		this.puesto = puesto;
	}

	public String getTipoEquipo() {
		return this.tipoEquipo;
	}

	public void setTipoEquipo(String tipoEquipo) {
		this.tipoEquipo = tipoEquipo;
	}

	public int getAula() {
		return this.aula;
	}

	public void setAula(int aula) {
		this.aula = aula;
	}
}

File: Creacion API/Python/equipo.py
class Equipo:
    def __init__(self, id, tipo_equipo, fecha_adquisicion, etiqueta, marca, modelo, descripcion, baja, aula_num, puesto):
        self.id = id
        self.tipo_equipo = tipo_equipo
        self.fecha_adquisicion = fecha_adquisicion
        self.etiqueta = etiqueta
        self.marca = marca
        self.modelo = modelo
        self.descripcion = descripcion
        self.baja = baja
        self.aula_num = aula_num
        self.puesto = puesto

    def get_id(self):
        return self.id

    def set_id(self, value):
        self.id = value
        
    def get_tipo_equipo(self):
        return self.tipo_equipo

    def set_tipo_equipo(self, value):
        self.tipo_equipo = value
    
    def get_fecha_adquisicion(self):
        return self.fecha_adquisicion

    def set_fecha_adquisicion(self, value):
        self.fecha_adquisicion = value
    
    def get_etiqueta(self):
        return self.etiqueta

    def set_etiqueta(self, value):
        self.etiqueta = value
        
    def get_marca(self):
        return self.marca

    def set_marca(self, value):
        self.marca = value
         
    def get_modelo(self):
        return self.modelo

    def set_modelo(self, value):
        self.modelo = value
        
    def get_descripcion(self):
        return self.descripcion

    def set_descripcion(self, value):
        self.descripcion = value
        
    def get_baja(self):
        return self.baja

    def set_baja(self, value):
        self.baja = value
        
    def get_aula_num(self):
        return self.aula_num

    def set_aula_num(self, value):
        self.aula_num = value
        
    def get_puesto(self):
        return self.puesto

    def set_puesto(self, value):
        self.puesto = value